File: Data Structure I/java/Day 14/ValidateBinarySearch.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        ArrayList<Integer> ans=new ArrayList<>();
        valid(root,ans);
        for(int i=1;i<ans.size();i++){
            if(ans.get(i)<=ans.get(i-1)){
                return false;
            }
        }
        return true;
    }
    public void valid(TreeNode root,ArrayList<Integer> ans){
        if(root==null){
            return;
        }
        valid(root.left,ans);
        ans.add(root.val);
        valid(root.right,ans);

    }
}

File: Data Structure I/python/Day 14/validate_binary_search.py
        ans=[]
        self.valid(root,ans)
        for i in range(len(ans)):
            if(ans[i]<=ans[i-1]):
                return false
        return true
    
    def valid(self,root,ans):
        if(root==None):
            return
        
        self.valid(root.left,ans)
        ans.append(root.val)
        self.valid(root.right,ans)

    